presto-main/src/main/java/com/facebook/presto/operator/aggregation/TypedSet.java

```diff
@@ -136,6 +136,26 @@ public boolean add(Block block, int position)
         return false;
     }
 
+    public int addAndGetPosition(Block block, int position)
```

Comment:
Can we create a common method between add and this method?

Reply:
no - we want to keep them separate for now so the checks are not expensive.

Reply:
We can keep the methods separate, but both these methods can use a common reusable method.

Reply:
done

Reply:
actually no - add returns status before and this one after. So let's keep them separate